### PR TITLE
Pr2 timezone

### PR DESCRIPTION
**Pull request by GALTdea**
Fixes issue #30 
Pr2:
-Adds params sanitation for the new attribute user time_zone. 
-Adds time_zone_select helper to devise_invitation and devise form

### CODE REVIEW

File: app/views/devise/invitations/edit.html.erb [hunk 1]
```diff
@@ -15,6 +15,12 @@
     </div>
   <% end %>
 
+  <div class=" mb-8">
+    <%= f.label :time_zone, class: 'label label-text' %>
+    <%= f.time_zone_select(:time_zone, ActiveSupport::TimeZone.country_zones('US'), {}, {class:"select select-bordered w-full max-w-xs"}) %>
```
Comment (toyhammered) [inline app/views/devise/invitations/edit.html.erb:20]: We want to allow all country codes. I think there are ways to prioritize some which might be useful?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,13 +2,22 @@ class ApplicationController < ActionController::Base
...
   private
   def set_time_zone(&)
     Time.use_zone(current_user.time_zone, &)
   end
+
+  protected
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:accept_invitation, keys: [:time_zone])
+    devise_parameter_sanitizer.permit(:account_update, keys: [:time_zone])
+  end
```
Comment (toyhammered) [inline app/controllers/application_controller.rb:22]: very random thing but protected is usually above private for readability reasons.

File: db/migrate/20230516173259_add_column_time_zone_to_users.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 class AddColumnTimeZoneToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :time_zone, :string, default: 'UTC', null: false
+  end
```
Comment (toyhammered) [inline db/migrate/20230516173259_add_column_time_zone_to_users.rb:4]: this is not right. This should be deleted

File: db/migrate/20230516173259_add_column_time_zone_to_users.rb [hunk 4]
```diff
@@ -8,4 +11,7 @@ def up
    def down
     remove_column :users, :time_zone
    end
+  def change
+    add_column :users, :time_zone, :string, default: 'UTC', null: false
+  end
```
Comment (toyhammered) [inline db/migrate/20230516173259_add_column_time_zone_to_users.rb:16]: same here. Looks like your merge had some issues.

**Review by toyhammered — CHANGES_REQUESTED** [review]
This looks great for the most part!

File: app/views/devise/invitations/edit.html.erb (hunk 1 above)
Comment (GALTdea) [reply app/views/devise/invitations/edit.html.erb:20]: This setting shows all timezones. the second option `ActiveSupport::TimeZone.country_zones('US')` only prioritizes the US time_zones, it places them on top.

**Review by toyhammered — APPROVED** [review]

**Pull request merged**